manifests/init.pp

```diff
@@ -87,6 +87,7 @@
   $defaults_options = $haproxy::params::defaults_options,
   $restart_command  = undef,
   $custom_fragment  = undef,
+  $config_file      = $haproxy::params::config_file,
```

Comment:
you allow people to override the config_file in haproxy, but then you don't use that parameter in sub-classes and sub-defined types? and instead still include ::haproxy::params, and use that config_file??